cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cmath>
+#include <cassert>
+#include <functional>
+#include <iostream>
+#include <limits>
+
+namespace cudf {
+namespace fixed_point {
+
+template <typename T>
+struct weak_typedef {
```

Comment:
This was the alternative to `BOOST_STRONG_TYPEDEF` which was the Boost dependency that we wanted to get rid of. It enable strong typing at the construction site of a `fixed_point` type for the `scale` parameter

Reply:
Then why is it called `weak_typedef`?

Reply:
Because it decays into the underlying type (ake `int32_t`) so its not really a **strong** typedef. It is only required at the construction of the type.

Reply:
Documentation. :)

Reply:
Will add.

Reply:
Done.